### PR TITLE
Add function to upload Python packages to Nexus

### PR DESCRIPTION
**Pull request by athos-ribeiro**
Signed-off-by: Athos Ribeiro <athos@redhat.com>

### CODE REVIEW

File: cachito/workers/nexus.py [hunk 1]
```diff
@@ -310,10 +311,16 @@ def upload_artifact(repo_name, repo_type, artifact_path):
     with open(artifact_path, "rb") as artifact:
         file_payload = {f"{repo_type}.asset": artifact.read()}
 
-    username, password = _get_nexus_hoster_credentials()
-    auth = requests.auth.HTTPBasicAuth(username, password)
-    url = f"{_get_nexus_hoster_url()}/service/rest/v1/components"
     config = get_worker_config()
+    if to_nexus_hoster:
+        username, password = _get_nexus_hoster_credentials()
+        url = f"{_get_nexus_hoster_url()}/service/rest/v1/components"
```
Comment (nirzari) [inline cachito/workers/nexus.py:317]: Would it be preferable to use `os.path.join()` for url concatenations?

File: cachito/workers/nexus.py [hunk 2]
```diff
@@ -310,10 +311,16 @@ def upload_artifact(repo_name, repo_type, artifact_path):
...
+    else:
+        username = config.cachito_nexus_username
+        password = config.cachito_nexus_password
+        url = f"{config.cachito_nexus_url}/service/rest/v1/components"
```
Comment (nirzari) [inline cachito/workers/nexus.py:321]: Same as above

**Comment by athos-ribeiro** [comment]
This is ready for review.

@nirzari I am requesting your review again because I made substantial changes to the code you had reviewed before.

File: cachito/workers/nexus.py [hunk 3]
```diff
@@ -295,46 +295,98 @@ def search_components(**query_params):
...
+    try:
+        upload_component(params, payload, to_nexus_hoster)
+    except CachitoError:
+        log.exception("Failed to upload %r to the %r Nexus repository", component_path, repo_type)
+        raise
+
+
+def upload_raw_component(repo_name, directory, components, to_nexus_hoster=True):
+    """
+    Upload an asset only component to a Nexus hosted repository.
```
Comment (chmeliik) [inline cachito/workers/nexus.py:327]: The description should probably be different, this seems to be copied from the other upload function.

**Review by chmeliik — COMMENTED** [review]
Looks great! Just one comment for a docstring.

File: cachito/workers/nexus.py (hunk 3 above)
Comment (athos-ribeiro) [reply cachito/workers/nexus.py]: Ha! You got me :see_no_evil: 
Fixed

**Comment by athos-ribeiro** [comment]
Still need to rebase this on master

**Comment by athos-ribeiro** [comment]
> Still need to rebase this on master

done

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Pull request merged**